File: CaseStudyPropertyTestsFs/ReverserPropertyTests.fs
module ReverserPropertyTests

open FsCheck
open NUnit.Framework
open CaseStudy

type R = Reverser

let myConfig = Config.VerboseThrowOnFailure

[<Test>]
let ``string reversed and then reversed again is same as original string``() =
    Check.One(myConfig, (fun s -> R.Reverse(R.Reverse(s)) = s));

File: CaseStudyPropertyTestsCs/ReverserPropertyTests.cs
using CaseStudy;
using FsCheck;
using Microsoft.FSharp.Core;
using NUnit.Framework;

namespace CaseStudyPropertyTestsCs
{
    using PropFunc = FSharpFunc<string, bool>;

    [TestFixture]
    public class ReverserPropertyTests
    {
        private static readonly Config MyConfig = Config.VerboseThrowOnFailure;

        [Test]
        public void StringReversedAndThenReversedAgainIsSameAsOriginalString()
        {
            Check.One(MyConfig, PropFunc.FromConverter(s => s.Reverse().Reverse() == s));
        }
    }
}
